cmd/wallet/main.go

```diff
@@ -1,163 +1,391 @@
 package main
 
 import (
-	"errors"
+	"encoding/binary"
+	"flag"
 	"fmt"
 	"io/fs"
+	"log"
 	"os"
 	"os/user"
 	"path/filepath"
+	"strings"
 
-	"github.com/howeyc/gopass"
 	"github.com/mr-tron/base58"
-	flag "github.com/spf13/pflag"
+
+	"github.com/pkg/errors"
+
+	"github.com/howeyc/gopass"
+	"github.com/tyler-smith/go-bip39"
+	"github.com/wavesplatform/gowaves/pkg/crypto"
+	"github.com/wavesplatform/gowaves/pkg/proto"
 	"github.com/wavesplatform/gowaves/pkg/wallet"
 )
 
-var usage = `
+const (
+	newOpt               = "new"
+	showOpt              = "show"
+	seedPhraseOpt        = "seed-phrase"
+	seedPhraseBase58Opt  = "seed-phrase-base58"
+	accountSeedBase58Opt = "account-seed-base58"
+
+	schemeOpt = "scheme"
+)
 
+const (
+	defaultBitSize    = 160
+	walletDefaultName = ".waves"
+)
+
+var usage = `
 Usage:
-  wallet command [flags]
+  wallet [flags]
 
-Available Commands:
-  add          Add seed to wallet
-  show         Print wallet data
+Flags:
+`
 
+var examples = `
+Examples:
+	./wallet -show -wallet <wallet path> // Show existing wallet credentials
+	./wallet -new // Generate a seed phrase and a wallet 
+	./wallet -seed-phrase "..." // Import a seed phrase
+	./wallet -seed-phrase-base58 "..." // Import a Base58 encoded seed phrase
+	./wallet -account-seed "..." // Import a Base58 encoded account seed
 `
 
 type Opts struct {
-	Force        bool
-	PathToWallet string
-	Base58       bool
+	PathToWallet       string
+	AccountNumber      int
+	Scheme             proto.Scheme
+	SeedPhrase         string
+	IsSeedPhraseBase58 bool
+	Base58AccountSeed  string
+	New                bool
+	Show               bool
+}
+
+type SchemeFlag byte
+
+func (sch *SchemeFlag) String() string {
+	str := string(*sch)
+	return str
+}
+func (sch *SchemeFlag) Set(s string) error {
+	val := []byte(s)
+	if len(val) > 1 {
+		return errors.New("failed to parse scheme: more than one letter was provided")
+	}
+
+	*sch = SchemeFlag(val[0])
+	return nil
 }
 
 func main() {
 	opts := Opts{}
-
-	flag.BoolVarP(&opts.Force, "force", "f", false, "Overwrite existing wallet")
-	flag.StringVarP(&opts.PathToWallet, "wallet", "w", "", "Path to wallet")
-	flag.BoolVarP(&opts.Base58, "base58", "b", false, "Input seed as Base58 encoded string")
+	var scheme SchemeFlag
+	flag.BoolVar(&opts.New, newOpt, false, "Generate and add a new seed phrase (Primary flag)")
+	flag.BoolVar(&opts.New, showOpt, false, "Show existing wallet credentials (Primary flag)")
+	flag.StringVar(&opts.SeedPhrase, seedPhraseOpt, "", "Import a seed phrase (Primary flag)")
+	flag.BoolVar(&opts.IsSeedPhraseBase58, seedPhraseBase58Opt, false, "Import a base58-encoded seed phrase (Primary flag)")
```

Comment:
Again, this flag that shows that given seed phrase is Base58 encoded is not primary. Because from one point of view, "primary flag" means that only one can be used at a time. Which is not the case for the `IsSeedPhraseBase58`. I think only `-new`,  `-seed-phrase` and `-account-seed` are mutually exclusive. If someone use `-seed-phrase-base58` with `-account-seed` it can be ignored, because account seed is always Base58 encoded.

Reply:
My fault, it's not supposed to be there, I thought I pushed the commit